src/pages/profiles/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import DefaultLayout from "../../layouts/default";
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
+import request from '../../utils/request';
+
+interface Profile {
+  _id: string;
+  name: string;
+  description?: string;
+  permissions: string[];
+}
+
+const fetchProfiles = async (): Promise<Profile[]> => {
+  const response = await request('/api/profiles');
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
+  }
+  return response.json();
```

Comment:
Não há tratamento de erro para response.json(), que pode falhar se a resposta não for um JSON válido.